libs/platform/src/lib/components/form/form-generator/interfaces/dynamic-form-item.ts

```diff
@@ -0,0 +1,159 @@
+import { AsyncValidatorFn, ValidatorFn } from '@angular/forms';
+import { Observable } from 'rxjs';
+import { SelectItem, ContentDensity, HintPlacement, LabelLayout, InputType } from '@fundamental-ngx/platform';
+
+export type DynamicFormItemChoices = number | string | SelectItem;
+export type DynamicFormItemValidationResult = null | boolean | string;
+
+/**
+ * Dynamic form item object interface which is used for generating form controls of defined type with validation rules.
+ */
+export interface DynamicFormItem {
+    /**
+     * @description
+     * Type of the form item. E.g. input, textarea, checkbox, etc.
+     */
+    type: string;
+
+    /**
+     * @description
+     * Additional set of options that can affect UI of the form item form control.
+     */
+    controlType?: InputType;
+
+    /**
+     * @description
+     * Name of the form item in form.
+     */
+    name: string;
+
+    /**
+     * @description
+     * Display name of the form item.
+     * @param formValue the form value hash.
+     */
+    message: string | ((formValue?: {[key: string]: any}) => string | Promise<string> | Observable<string>);
```

Comment:
```suggestion
    interface FormValue<T> {
        [key: string]: T
    }
    message: string | ((formValue?: FormValue) => string | Promise<string> | Observable<string>);
```

Reply:
could you please create an interface for formValue and use it everywhere?

Reply:
`formValue` is a hash object where values might be of different types, 
the proposed type constrains us to have all values of the same type, it is not valid for this case